internal/provider/resource_rediscloud_active_active_region.go

```diff
@@ -0,0 +1,517 @@
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"vpc_id": {
+							Description: "Identifier of the VPC to be peered",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"recreate_region": {
+							Description: "Defines wheter the regions should be re-created",
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+						},
+						"networking_deployment_cidr": {
+							Description:      "Deployment CIDR mask",
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateDiagFunc: validateDiagFunc(validation.IsCIDR),
+						},
+
+						"database": {
+							Description: "TODO",
+							Type:        schema.TypeSet,
+							Required:    true,
+							MinItems:    1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Description: "A numeric id for the database",
+										Type:        schema.TypeInt,
+										Required:    true,
+									},
+									"database_name": {
+										Description: "A meaningful name to identify the database",
+										Type:        schema.TypeString,
+										Required:    true,
+									},
+									"local_write_operations_per_second": {
+										Description: "Write operations per second for creation plan databases",
+										Type:        schema.TypeInt,
+										Required:    true,
+									},
+									"local_read_operations_per_second": {
+										Description: "Write operations per second for creation plan databases",
+										Type:        schema.TypeInt,
+										Required:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceRedisCloudActiveActiveRegionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*apiClient)
+	var diags diag.Diagnostics
+
+	subId, err := strconv.Atoi(d.Get("subscription_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Get existing regions so we can do a manual diff
+	// Query API for existing Regions for a given Subscription
+	existingRegions, err := api.client.Regions.List(ctx, subId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Create an existingRegionMap<regionName, region>
+	existingRegionMap := make(map[string]*regions.Region)
+	for _, existingRegion := range existingRegions.Regions {
+		existingRegionMap[*existingRegion.Region] = existingRegion
+	}
+
+	regionsFromResourceData := buildCreateActiveActiveRegions(d.Get("region").(*schema.Set))
+
+	err, _ = regionCreate(subId, existingRegionMap, regionsFromResourceData, ctx, d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func regionCreate(subId int, existingRegionMap map[string]*regions.Region, regionsFromResourceData []*regions.Region, ctx context.Context, d *schema.ResourceData, meta interface{}) (error, []*regions.Region) {
+	api := meta.(*apiClient)
```

Comment:
I will only mention here, but instead of sending through the meta and then casting it, I would just send through the client type.